Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.py

```diff
@@ -0,0 +1,658 @@
+    if severity:
+        filters.append({"field": "severity", "operator": "in", "value": severity})
+
+    request_data = {'request_data': {'filters': filters, 'search_from': 0,
+                                     'search_to': max_fetch, 'sort': {'field': 'creation_time', 'keyword': 'asc'}}}
+
+    raw = client.list_alerts_request(request_data)
+
+    items = raw.get('reply', {}).get('alerts')
+    for item in items:
+        # for item in items.outputs:
+        incident_created_time = item['detection_timestamp']
+        incident = {
+            'name': item['name'],
+            'details': item['description'],
+            'occurred': timestamp_to_datestring(incident_created_time),
+            'rawJSON': json.dumps(item),
+            'severity': convert_to_demisto_severity(item.get('severity', 'Low'))
+        }
+
+        incidents.append(incident)
+
+        if incident_created_time > latest_created_time:
+            latest_created_time = incident_created_time
+
+    next_run = {'last_fetch': latest_created_time}
+    return next_run, incidents
+
+
+def test_module(client: Client) -> None:
+    """
+    Tests API connectivity and authentication'
+    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
+    successful.
+    Raises exceptions if something goes wrong.
+
+    Args:
+        client (Client): CortexXpanse client to use.
+
+    Returns:
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+    try:
+        client.list_external_service_request([])
+    except DemistoException as e:
+        if 'Forbidden' in str(e):
+            raise DemistoException('Authorization Error: make sure API Key is correctly set')
+        else:
+            raise e
+    return_results('ok')
+
+
+def main() -> None:
+    """
+    main function
+    """
+    params: Dict[str, Any] = demisto.params()
+    args: Dict[str, Any] = demisto.args()
+
+    command = demisto.command()
+    demisto.debug(f'Command being called is {command}')
+
+    try:
+        first_fetch_time = arg_to_datetime(
+            arg=params.get('first_fetch', '3 days'),
+            arg_name='First fetch time',
+            required=True
+        )
+        first_fetch_timestamp = int(first_fetch_time.timestamp()) * 1000 if first_fetch_time else None
+        severity = params.get('severity')
+        max_fetch = int(params.get('max_fetch', 10))
+        creds = params.get('credentials', {})
+        api = creds.get('password', '')
+        auth_id = creds.get('identifier', '')
+        headers = {
+            'Authorization': f'{api}',
+            'x-xdr-auth-id': f'{auth_id}',
+            'Content-Type': 'application/json'
+        }
+
+        proxy = params.get('proxy', False)
+        handle_proxy()
+        verify_certificate = not params.get('insecure', False)
+
+        url_suffix = "/public_api/v1"
```

Comment:
please set `url_suffix` as constant variable and  move it to the top of the code

Reply:
complete

Reply:
[adi recommendations](https://github.com/demisto/content/pull/23592/commits/4e5a2f1e787a6dbddfd39f955ab80c0564c8bca0)